__tests__/api/ExperimentsApi.test.ts

```diff
@@ -191,6 +193,66 @@ describe('ExperimentsApi.ts module', () => {
     })
   })
 
+  describe('assignMetric', () => {
+    it('should make the right request', async () => {
+      // NOTE: We are unit testing this one as the request it makes isn't so simple
+      const experiment = Fixtures.createExperimentFull()
+
+      // @ts-ignore
+      const origFetch = global.fetch
+      // @ts-ignore
+      global.fetch = jest.fn().mockImplementationOnce(async () => ({
+        json: async () => experiment,
+      }))
+
+      const newMetricAssignment = Fixtures.createMetricAssignment({}) as MetricAssignmentNew
+      // @ts-ignore
+      newMetricAssignment.metricAssignmentId = undefined
```

Comment:
Can't we do something like this?

```suggestion
      const newMetricAssignment = Fixtures.createMetricAssignment({metricAssignmentId: undefined}) as MetricAssignmentNew
```